tests/client/amo/components/TestAddonDetail.js

```diff
@@ -291,19 +298,27 @@ describe('AddonDetail', () => {
         ...fakeAddon,
         type: ADDON_TYPE_THEME,
       },
-      clientSupportsAddons: () => true,
+      getCompatibleClient: () => ({ compatible: true }),
     });
     const button = rootNode.querySelector('.AddonDetail-theme-header-label');
     assert.equal(button.disabled, false);
   });
 
+  it('disables install switch for unsupported clients', () => {
+    const rootNode = renderAsDOMNode({
+      getClientCompatibility: getClientCompatibilityFalse,
+    });
+    assert.isTrue(
+      rootNode.querySelector('.InstallButton-switch input').disabled);
+  });
+
   it('disables a theme preview for unsupported clients', () => {
     const rootNode = renderAsDOMNode({
       addon: {
         ...fakeAddon,
         type: ADDON_TYPE_THEME,
       },
-      clientSupportsAddons: () => false,
+      getClientCompatibility: getClientCompatibilityFalse,
```

Comment:
Well, `<AddonCompatibilityError />` throws an error if any of the required props are missing, so yes. I had to go back and fix some tests that had them missing. I can add an explicit one where `getClientCompatibility()` doesn't return all required props and throws. It's largely covered elsewhere already though, if I understand correctly.